includes/models/class-wc-stripe-subscription.php

```diff
@@ -0,0 +1,298 @@
+		if ( $subscription instanceof WC_Stripe_Subscription ) {
+			$this->wc_subscription = $subscription->wc_subscription;
+		} elseif ( class_exists( 'WC_Subscription' ) && $subscription instanceof WC_Subscription ) {
+			$this->wc_subscription = $subscription;
+		}
+
+		if ( ! $this->wc_subscription ) {
+			$type = is_object( $subscription ) ? get_class( $subscription ) : gettype( $subscription );
+			throw new InvalidArgumentException( 'WC_Stripe_Subscription requires a valid underlying WC_Subscription; supplied $subscription argument was ' . $type );
+		}
+	}
+
+	/**
+	 * Magic method to call methods on the underlying WC_Subscription object.
+	 *
+	 * @param string $name The name of the method to call.
+	 * @param array $arguments The arguments to pass to the method.
+	 */
+	public function __call( $name, $arguments ) {
+		if ( method_exists( 'WC_Subscription', $name ) ) {
+			return call_user_func_array( [ $this->wc_subscription, $name ], $arguments );
+		}
+
+		throw new BadMethodCallException( 'Method ' . $name . ' does not exist in WC_Stripe_Subscription' );
+	}
+
+	/**
+	 * Wrapper to return an order using the extension's custom WC_Stripe_Subscription class.
+	 *
+	 * @param $subscription_id int Subscription ID.
+	 * @return bool|WC_Stripe_Subscription
+	 */
+	public static function get_by_id( $subscription_id ) {
+		$subscription = function_exists( 'wcs_get_subscription' ) ? wcs_get_subscription( $subscription_id ) : false;
+		return $subscription ? self::to_instance( $subscription ) : false;
+	}
+
+	/**
+	 * Wrapper to get subscriptions using the extension's custom WC_Stripe_Subscription class.
+	 *
+	 * @param $args array Arguments to pass to wcs_get_subscriptions.
+	 * @return array|WC_Stripe_Subscription[]
+	 */
+	public static function query( $args ) {
+		$subscriptions = function_exists( 'wcs_get_subscriptions' ) ? wcs_get_subscriptions( $args ) : [];
+		if ( empty( $subscriptions ) ) {
+			return [];
+		}
+
+		return array_map(
+			function ( $subscription ) {
+				return self::to_instance( $subscription );
+			},
+			$subscriptions
+		);
+	}
+
+	/**
+	 * Wrapper to get subscriptions for an order using the extension's custom WC_Stripe_Subscription class.
+	 *
+	 * @param $order int|WC_Order Order object or ID.
+	 * @return array|WC_Stripe_Subscription[]
+	 */
+	public static function get_for_order( $order ) {
+		$subscriptions = function_exists( 'wcs_get_subscriptions_for_order' ) ? wcs_get_subscriptions_for_order( $order ) : [];
+		if ( empty( $subscriptions ) ) {
+			return [];
+		}
+
+		return array_map(
+			function ( $subscription ) {
+				return self::to_instance( $subscription );
+			},
+			$subscriptions
+		);
+	}
+
+	/**
+	 * Wrapper to get subscriptions for a renewal order using the extension's custom WC_Stripe_Subscription class.
+	 *
+	 * @param $order int|WC_Order Order object or ID.
+	 * @return array|WC_Stripe_Subscription[]
+	 */
+	public static function get_for_renewal_order( $order ) {
```

Comment:
Another naming suggestion to mirror the underlying `wcs_get_subscriptions_for_renewal_order()` function.
```suggestion
	public static function get_subscriptions_for_renewal_order( $order ) {
```

Reply:
See #discussion_r2071631006